GNSSLogger/app/src/main/java/com/google/android/apps/location/gps/gnsslogger/MainActivity.java

```diff
@@ -148,7 +150,13 @@ protected void onCreate(Bundle savedInstanceState) {
 
   protected PendingIntent createActivityDetectionPendingIntent() {
     Intent intent = new Intent(this, DetectedActivitiesIntentReceiver.class);
-    return PendingIntent.getBroadcast(this, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+    // Missing mutability is only broken on S+, but immutable is available on M+,
+    // so it is good for M+ to set this property.
+    if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
+      return PendingIntent.getBroadcast(this, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT | FLAG_IMMUTABLE);
```

Comment:
I believe this needs to use `FLAG_MUTABLE` instead.

See:
* https://stackoverflow.com/questions/69862244/android-activityrecognition-and-immutable-pendingintent
* https://developer.android.com/guide/components/intents-filters#CreateImmutablePendingIntents

Reply:
> I believe this needs to use `FLAG_MUTABLE` instead on Android S and higher.
> 
> See:
> 
> * https://stackoverflow.com/questions/69862244/android-activityrecognition-and-immutable-pendingintent
> * https://developer.android.com/guide/components/intents-filters#CreateImmutablePendingIntents

Thank you very much. You are right, I've fixed it in the new commit.